oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/OneandoneComputeServiceAdapter.java

```diff
@@ -0,0 +1,214 @@
+   protected Logger logger = Logger.NULL;
+
+   private final OneAndOneApi api;
+   private final Predicate<Server> waitServerUntilAvailable;
+   private final List<DataCenter> datacetners;
+
+//   private static final Integer DEFAULT_LAN_ID = 1;
+   @Inject
+   OneandoneComputeServiceAdapter(OneAndOneApi api,
+           @Named(POLL_PREDICATE_SERVER) Predicate<Server> waitServerUntilAvailable) {
+      this.api = api;
+      this.datacetners = ImmutableList.of();
+      this.waitServerUntilAvailable = waitServerUntilAvailable;
+   }
+
+   @Override
+   public NodeAndInitialCredentials<Server> createNodeWithGroupEncodedIntoName(String group, String name, Template template) {
+      checkArgument(template.getLocation().getScope() == LocationScope.REGION, "Template must use a REGION-scoped location");
+
+      final String dataCenterId = template.getLocation().getId();
+      Hardware hardware = template.getHardware();
+      TemplateOptions options = template.getOptions();
+
+      final String loginUser = isNullOrEmpty(options.getLoginUser()) ? "root" : options.getLoginUser();
+      final String password = options.hasLoginPassword() ? options.getLoginPassword() : Passwords.generate();
+      final org.jclouds.compute.domain.Image image = template.getImage();
+
+      //prepare hdds to provision
+      List<? extends Volume> volumes = hardware.getVolumes();
+      List<Hdd.CreateHdd> hdds = new ArrayList<Hdd.CreateHdd>();
+
+      int i = 1;
+      for (final Volume volume : volumes) {
+         try {
+            Hdd.CreateHdd hdd = Hdd.CreateHdd.create(volume.getSize(), volume.isBootDevice());
+            hdds.add(hdd);
+         } catch (Exception ex) {
+            throw Throwables.propagate(ex);
+
+         }
+      }
+
+      // provision server
+      final Server server;
+      Double cores = ComputeServiceUtils.getCores(hardware);
+
+      try {
+         List<? extends Processor> processors = hardware.getProcessors();
+         org.apache.jclouds.oneandone.rest.domain.Hardware.CreateHardware hardwareRequest
+                 = org.apache.jclouds.oneandone.rest.domain.Hardware.CreateHardware.create(processors.size(), cores, hardware.getRam(), hdds);
+         final Server.CreateServer serverRequest = Server.CreateServer.builder()
+                 .name(name)
+                 .hardware(hardwareRequest)
+                 .applianceId(image.getId())
+                 .dataCenterId(dataCenterId)
+                 .password(password)
+                 .powerOn(Boolean.TRUE).build();
+
+         logger.trace("<< provisioning server '%s'", serverRequest);
+
+         server = api.serverApi().create(serverRequest);
+         logger.trace(">> provisioning complete for server. returned id='%s'", server.id());
+
+      } catch (Exception ex) {
+         logger.error(ex, ">> failed to provision server. rollbacking..");
+         throw Throwables.propagate(ex);
+      }
+      waitServerUntilAvailable.apply(server);

```

Comment:
There is no need to wait here. Jclouds will already wait for the server to be running, so let's better remove this.